src/widgets/visit/visit-button-component.tsx

```diff
@@ -181,10 +181,12 @@ const closeActiveVisitConfirmation = (currentVisit: any): React.ReactNode => {
   );
 };
 
-const endVisit = (currentVisit: any): React.ReactNode => {
+export const endVisit = (currentVisit: any): React.ReactNode => {
```

Comment:
Any chance of having a real React component here, e.g.,

```ts
export const EndVisit: React.FC<EndVisitProps> = ({ currentVisit }) => (
  ...
);
```

where

```ts
export interface EndVisitProps {
  currentVisit: VisitData;
}
```

(notice the `VisitData`, I don't know if we have a proper type for this already, but it should not be `any`)

Reply:
Yes, I have started working on it to make a React component.

Reply:
Hey @FlorianRappl , we have Visit for VisitData. I had to change the startDatetime and stopDatetime type from string to Date as it was conflicting between components. I have created a React component for endVisit.